File: jwt/src/main/java/com/member/jwt/controller/MemberController.java
package com.member.jwt.controller;

import com.member.jwt.dto.MemberInfoDto;
import com.member.jwt.dto.MemberUpdateDto;
import com.member.jwt.service.MemberService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/auth")
public class MemberController {

    private final MemberService memberService;

    public MemberController(MemberService memberService) {
        this.memberService = memberService;
    }

    @GetMapping
    public ResponseEntity<MemberInfoDto> getMemberInfo(@RequestParam Integer memberId) {
        MemberInfoDto memberInfo = memberService.getMemberInfo(memberId);
        return ResponseEntity.ok(memberInfo);
    }

    @PatchMapping
    public ResponseEntity<MemberInfoDto> updateMemberInfo(
            @RequestParam Integer memberId,
            @ModelAttribute MemberUpdateDto memberUpdateDto) {
        MemberInfoDto updatedMemberInfo = memberService.updateMemberInfo(memberId, memberUpdateDto);
        return ResponseEntity.ok(updatedMemberInfo);
    }

    @DeleteMapping
    public ResponseEntity<String> deleteMember(@RequestParam Integer memberId) {
        memberService.deleteMemberInfo(memberId);
        return ResponseEntity.ok("회원 삭제 완료");
    }
}

File: jwtKotlin/src/main/kotlin/com/member/jwtkotlin/controller/MemberController.kt
package com.member.jwtkotlin.controller

import com.member.jwtkotlin.dto.MemberInfoDto
import com.member.jwtkotlin.dto.MemberUpdateDto
import com.member.jwtkotlin.service.MemberService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/auth")
class MemberController(private val memberService: MemberService) {
    @GetMapping
    fun getMemberInfo(@RequestParam memberId: Int?): ResponseEntity<MemberInfoDto> {
        val memberInfo = memberService.getMemberInfo(memberId!!)
        return ResponseEntity.ok(memberInfo)
    }

    @PatchMapping
    fun updateMemberInfo(
        @RequestParam memberId: Int?,
        @ModelAttribute memberUpdateDto: MemberUpdateDto?
    ): ResponseEntity<MemberInfoDto> {
        val updatedMemberInfo = memberService.updateMemberInfo(memberId!!, memberUpdateDto!!)
        return ResponseEntity.ok(updatedMemberInfo)
    }

    @DeleteMapping
    fun deleteMember(@RequestParam memberId: Int?): ResponseEntity<String> {
        memberService.deleteMemberInfo(memberId!!)
        return ResponseEntity.ok("회원 삭제 완료")
    }
}